[Changelog] Add SE-0253: `func callAsFunction`.

dan-zheng opened the pull request:
Changelog entry for [SE-0253](https://github.com/apple/swift-evolution/blob/master/proposals/0253-callable.md).
Follow-up to implementation: https://github.com/apple/swift/pull/24299.

dan-zheng committed:
[Changelog] Add SE-0253: `func callAsFunction`.

karwa commented on CHANGELOG.md:
```diff
@@ -26,6 +26,27 @@ CHANGELOG
+    }
+  }
+  var adder = Adder(base: 3)
+  adder(10) // same as `adder.callAsFunction(10)`
```
Warning: unused return value. Might be nice to point out that this returns 13.

dan-zheng committed:
Clarify `callAsFunction` result.

dan-zheng committed:
Add more features.

Add `throws`/`rethrows` and trailing closures to feature list.
Features are tested in test/Sema/call_as_function_simple.swift.

dan-zheng committed:
Use a list for clearer formatting.

lattner reviewed (approved):
Awesome!

dan-zheng commented:
@swift-ci Please test

dan-zheng commented:
Seems like there was an unrelated failure:
```
/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swiftpm/Tests/FunctionalTests/MiscellaneousTests.swift:62: error: MiscellaneousTestCase.testCanBuildMoreThanTwiceWithExternalDependencies : failed - `swift build -c Release' failed:
14:55:03 
14:55:03 executionFailure(error: terminated(1): /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/buildbot_linux/swiftpm-linux-x86_64/x86_64-unknown-linux/release/swift-build --package-path /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/app --configuration release output:
14:55:03     Fetching /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/deck-of-playing-cards
14:55:03     Fetching /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/PlayingCard
14:55:03     Fetching /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/FisherYates
14:55:03     Cloning /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/deck-of-playing-cards
14:55:03     Resolving /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/deck-of-playing-cards at 1.2.3
14:55:03     Cloning /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/FisherYates
14:55:03     Resolving /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/FisherYates at 1.2.3
14:55:03     Cloning /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/PlayingCard
14:55:03     Resolving /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/PlayingCard at 1.2.3
14:55:03     [1/3] Compiling FisherYates Fisher-Yates_Shuffle.swift
14:55:03     <unknown>:0: warning: missing submodule 'SwiftOverlayShims'
14:55:03 
    /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/tmp/DependencyResolution_External_Complex.8xO2YG/app/.build/checkouts/FisherYates/src/Fisher-Yates_Shuffle.swift:4:8: error: missing required module 'SwiftOverlayShims'
14:55:03     import Glibc
```

dan-zheng commented:
@swift-ci Please test Linux

dan-zheng commented:
@swift-ci Please test Linux
